pandas/_libs/buffer.pyx

```diff
@@ -0,0 +1,29 @@
+from cpython cimport Py_buffer
+from libc.stdint cimport (
+    uint8_t,
+    uintptr_t,
+)
+
+
+cdef class CBuffer:
+    def __getbuffer__(self, Py_buffer *buffer, int flags):
+        cdef Py_ssize_t itemsize = sizeof(uint8_t)
+        cdef uintptr_t ptr = self.ptr
+        cdef Py_ssize_t[1] shape = tuple((self.bufsize // itemsize,))
+        cdef Py_ssize_t[1] strides = tuple((itemsize,))
+
+        buffer.buf = <void*>ptr
```

Comment:
There might be a better way of doing this overall, but I figured since `self.ptr` returns an int it is an easy bolt on to the existing design